File: RAnalytics/Util/Categories/UIApplication+Additions.m
#import "UIApplication+Additions.h"
#import "_RAnalyticsHelpers.h"

@implementation UIApplication(Additions)

+ (BOOL)_rat_respondsToSharedApplication
{
    return _RAnalyticsSharedApplication() != nil;
}

+ (UIInterfaceOrientation)_rat_statusBarOrientation
{
    return [_RAnalyticsSharedApplication() statusBarOrientation];
}

@end

File: RAnalytics/Util/Private/UIApplication+Additions.swift
import UIKit

@objc public extension UIApplication {
    static var ratStatusBarOrientation: UIInterfaceOrientation {
        guard let statusBarOrientation = UIApplication.RAnalyticsSharedApplication?.statusBarOrientation else {
            // [UIApplication sharedApplication] is not available for App Extension
            return .portrait // default returned value
        }
        return statusBarOrientation
    }
}
